### PR TITLE
Update downgrade-upgrade test

### PR DESCRIPTION
**Pull request by bitdivine**
# Motivation
The downgrade-upgrade test needs to be able to verify that accounts are preserved across schema updates.

# Changes
- Make it possible to specify a wasm to downgrade to.  This way the test can be run with a wasm that implements the future schema `S0.early` and another, `undefined.late` that is not yet in production.
- Get account data; this was already present but commented out as prod did not have account data in a stable, usable form.

# Tests

# Todos

- [x] Add entry to changelog (if necessary).

### CODE REVIEW

File: .config/spellcheck.dic [hunk 1]
```diff
@@ -5,6 +5,7 @@ ICP
 schemas
 WASM
 Wasm
+Wasms
```
Comment (bitdivine) [inline .config/spellcheck.dic:8]: Plural of Wasm.  By the way, technically, [WASM is incorrect according to the Wasm org](https://webassembly.org/) but we use it widely and that is not the worst issue we have with text.

File: scripts/nns-dapp/downgrade-upgrade-test [hunk 2]
```diff
@@ -32,9 +31,16 @@ get_current_wasm() {
 }
 
 get_prod_wasm() {
-  # TODO: Use the test build once available for the prod release.
   # Note: Using the test build allows us to get arbitrary test accounts from the canister.
-  curl -sL https://github.com/dfinity/nns-dapp/releases/download/prod/nns-dapp.wasm.gz >"$PROD_WASM"
+  if test -n "${PROD_WASM:-}"; then
+    test -f "${PROD_WASM}" || {
```
Comment (lmuntaner) [inline scripts/nns-dapp/downgrade-upgrade-test:36]: Sorry my ignorance, but it's hard to find on the internet with this name.

What does the "test" command do? Check for presence?

**Review by lmuntaner — APPROVED** [review]
LGTM, thanks!

File: scripts/nns-dapp/downgrade-upgrade-test (hunk 2 above)
Comment (bitdivine) [reply scripts/nns-dapp/downgrade-upgrade-test:36]: `test` is a bash builtin for evaluating [an expression](https://www.gnu.org/software/bash/manual/bash.html#Bash-Conditional-Expressions).  E.g. `-n` checks whether the string is non-empty and `-f` checks whether the path exists and is  a file.

**Pull request merged**